### PR TITLE
[ops, perf]: Add npu cross entropy fast path to accelerate chunk logp computation in RL scenarios

### PR DESCRIPTION
**Pull request by mikequan0425**
### What does this PR do?

> Concise overview of the change. Reference related issues/PRs.

As title.

### Checklist Before Starting

- Search for relative PRs/issues and link here: https://github.com/ByteDance-Seed/VeOmni/pulls?q=is%3Apr+is%3Aopen+NPU+cross+entropy
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))
  - **Allowed modules:** `agent`, `ci`, `ckpt`, `config`, `data`, `dist`, `docker`, `docs`, `logging`, `lora`, `misc`, `model`, `omni`, `optim`, `ops`, `parallel`, `perf`, `release`, `task`, `trainer`
  - **Allowed types:** `feat`, `fix`, `refactor`, `chore`, `test`
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> Validation results (training curves, eval metrics) for changes not covered by CI.

Run `python tests/ops/test_npu_chunk_logprobs.py --device npu --dtype float32`

image

Results in deferent chunksize. I manually adjusted the chunksize to investigate performance under various scenarios. The left graph corresponds to 3072, while the right graph uses the default value of 1024:

In the left figure, the red line represents the original method and the purple line stands for the method with NPU operators replaced;
In the right figure, the yellow line represents the original method and the red line stands for the method with NPU operators replaced.
image

### API and Usage Example

> Show API changes and usage examples if applicable.

It will automatically detect whether the device is NPU and then perform replacement.

### Design & Code Changes

> High-level design description and specific change list.

image

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

**Comment by mikequan0425** [comment]
@Luosuu @FoolPlayer Could you help review this PR, thanks.
